src/interpreter/internal.rs

```diff
@@ -166,10 +166,41 @@ impl<S> Interpreter<S> {
         }?;
 
         // update the local copy of the output
-        let mut output = Output::variable(owner_to_set, amount_to_set, asset_id_to_update);
+        let output = Output::variable(owner_to_set, amount_to_set, asset_id_to_update);
+        self.set_output(out_idx, output)
+    }
+
+    pub(crate) fn revert_variable_output(&mut self, out_idx: usize) -> Result<(), RuntimeError> {
+        if out_idx >= self.tx.outputs().len() {
+            return Err(PanicReason::OutputNotFound.into());
```

Comment:
Hmmm. Is `PanicReason` not for a VM panic? Presumably if this condition is met then the Rust code is buggy, no? The VM itself doesn't have the functionality to revert variable outputs.

This should be an `InterpreterError` rather than a `RuntimeError`.

Reply:
What @adlerjohn said seems to be correct. `PanicReason` is expected to be a result of valid transactions, and in this case it would be a result of an invalid and/or incorrect execution.